extensions/wikia/AdEngine/js/slot/scrollHandler.js

```diff
@@ -1,67 +1,110 @@
 /*global define*/
 define('ext.wikia.adEngine.slot.scrollHandler', [
-    'ext.wikia.adEngine.adContext',
-    'ext.wikia.adEngine.adHelper',
-    'wikia.log',
-    'wikia.document',
-    'wikia.window'
+	'ext.wikia.adEngine.adContext',
+	'ext.wikia.adEngine.adHelper',
+	'wikia.log',
+	'wikia.document',
+	'wikia.window'
 ], function (adContext, adHelper, log, doc, win) {
-    'use strict';
+	'use strict';
+	var logGroup = 'ext.wikia.adEngine.slot.scrollHandler',
+		isRefreshed = {},
+		reloadedView = {},
+		context = adContext.getContext(),
+		config = context.opts.scrollHandlerConfig || {
+			oasis: {
+				PREFOOTER_LEFT_BOXAD: {
+					reloadedViewMax: 1
+				},
+				PREFOOTER_RIGHT_BOXAD: {
+					reloadedViewMax: 3
+				}
+			},
+			mercury: {
+				MOBILE_PREFOOTER: {
+					reloadedViewMax: 1
+				}
+			}
+		},
+		skin = context.targeting.skin === 'oasis' ? 'oasis' : 'mercury';
 
-    var logGroup = 'ext.wikia.adEngine.slot.scrollHandler',
-        isRefreshed = {
-            PREFOOTER_LEFT_BOXAD: false,
-            PREFOOTER_RIGHT_BOXAD: false
-        },
-        reloadedView = {
-            PREFOOTER_LEFT_BOXAD: 0,
-            PREFOOTER_RIGHT_BOXAD: 0
-        };
+	function init() {
+		if (context.opts.enableScrollHandler)  {
+			config = config[skin];
+			prepareSettings();
+			registerSlotEvents();
+		}
+	}
 
-    function init() {
-        if (adContext.getContext().opts.enableScrollHandler) {
-            win.addEventListener('scroll', adHelper.throttle(function () {
-                log('Scroll event listener has been added', 'debug', logGroup);
-                for (var slotName in isRefreshed) {
-                    if (isRefreshed.hasOwnProperty(slotName)) {
-                        refreshSlot(slotName);
-                    }
-                }
-            }));
-        }
-    }
+	function prepareSettings() {
+		for (var slotName in config) {
+			if (config.hasOwnProperty(slotName)) {
+				isRefreshed[slotName] = false;
+				reloadedView[slotName] = 0;
+			}
+		}
+	}
 
-    function refreshSlot(slotName) {
-        var status = isReached(doc.getElementById(slotName));
-        if (!isRefreshed[slotName] && status) {
-            log(['refreshSlot', slotName + ' has been refreshed'], 'debug', logGroup);
-            reloadedView[slotName]++;
-            win.adslots2.push(slotName);
-            isRefreshed[slotName] = true;
-        } else if (!status) {
-            isRefreshed[slotName] = false;
-        }
-    }
+	function registerSlotEvents() {
+		win.addEventListener('scroll', adHelper.throttle(function () {
+			log('Scroll event listener has been added', 'debug', logGroup);
+			for (var slotName in config) {
+				if (config.hasOwnProperty(slotName)) {
+					onScroll(slotName);
+				}
+			}
+		}));
+	}
 
-    function isReached(el) {
-        return win.innerHeight + win.scrollY >= getTopPos(el);
-    }
+	function refreshSlot(slotName) {
+		if (skin === 'oasis') {
+			win.adslots2.push([slotName]);
+		} else {
```

Comment:
Since skin is set to `oasis` or `mercury` and it's initialised only in `desktop.run.js` and `mercury.run.js` I think we don't need this else :P